rfcs/058-relevance-testing/examples.md

```diff
@@ -0,0 +1,101 @@
+# Examples of rank CLI usage
+
+All flags in the following commands should be optional, and the CLI should prompt the user for any missing information, allowing them to select from the available options where appropriate.
+
+```
+rank
+├── index
+│   ├── list
+│   ├── create
+│   ├── update
+│   ├── delete
+│   ├── get
+│   └── ccr
+├── task
+│   ├── check
+│   └── delete
+├── search
+│   ├── terms
+│   └── compare
+└── test
+    ├── run
+    └── list
+```
+
+```bash
+rank index list
+```
+
+List all the indices in the rank cluster.
+
+```bash
+rank index create --source <source> --target <target> --config <config>
+```
+
+Create a new index in the rank cluster, using data from a copy of a production index (also in the rank cluster), and a locally defined settings/mapping config.
+
+```bash
+rank index update --index <index> --config <config> --run
+```
+
+Update the settings/mapping config for an existing index in the rank cluster. Should allow the user to run the [update-by-query](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update-by-query.html) for all documents immediately, or leave it to be run later.
+
+```bash
+rank index delete --index <index>
+```
+
+Delete an index in the rank cluster.
+
+```bash
+rank index get --index <index>
+```
+
+Fetch a copy of the settings/mapping config for an index in the rank cluster.
+
+```bash
+rank index ccr --source <source> --target <target>
+```
+
+Copy an index from the rank cluster to a production cluster using [cross-cluster replication](https://www.elastic.co/guide/en/elasticsearch/reference/current/ccr-what-is.html).
+
+```bash
+rank task check --task <task>
+```
+
+Check the progress/status of a reindex or update-by-query task.
+
+```bash
+rank task delete --task <task>
+```
+
+Cancel a task.
+
+```bash
+rank search --index <index> --query <query> --search-terms <searchTerms>
+```
+
+Run a search against an index in the rank cluster, outputting formatted results to stdout.
+
+```bash
+rank search terms 
+```
+
+Fetch a set of real search terms from the reporting cluster for each content type, and write them to a local file.
```

Comment:
realised that i never responded to the second half of this comment!

We currently use a script to fetch 5000 recent search terms from the reporting cluster. they're intended to measure and compare the query speed distribution, so getting a random sample of real queries is useful!

Reply:
Does it matter that this means the profiling isn't strictly reproducible (unless you have the file of terms)?

Reply:
> unless you have the file of terms

Maybe we should be shoving the file into an S3 bucket or similar shared source instead?

Reply:
The hope is that that the _result_ of the test should be the same from run to run, regardless of the input data. Even if the granular inputs are different, the large sample of terms should lead to a well-matched distribution of query complexity and provide a good indication of how the query performs.

That said, I do think full reproducibility is worth striving for where we can. I think we could achieve the same result without adding an extra s3 bucket, because the reporting cluster is already a shared resource. If we allow users to specify a timestamp flag for the `get-terms` command, anyone would be able to fetch the same 5,000 terms from the search logs, eg `rank search get-terms --timestamp=2023-06-19T13:36:28`